app/controllers/samples_controller.rb

```diff
@@ -1426,4 +1428,21 @@ def check_owner
       # Rendering halts the filter chain
     end
   end
+
+  def warn_if_large_bulk_upload(samples)
+    # 100 is the limit that was chosen during covid19 surge for biohub projects.
+    # At the time, we had successfully handled up to ~1000 uploads per day.
+    if samples.length < 100
+      return
+    end
+    # assume that all samples are in the same project and from the same user
+    project = samples.first.project
+    uploader = samples.first.user
+    msg = "[Datadog] LargeBulkUploadEvent: #{samples.length} samples by #{uploader.role_name}." \
+      " See: #{project.status_url}"
+    LogUtil.log_err_and_airbrake(msg)
+  rescue StandardError => e
+    # catch all errors because we don't want to ever block uploads
+    LogUtil.log_err_and_airbrake("warn_if_large_bulk_upload: #{e}")
```

Comment:
i think i forgot this on the other PR. I think you need to use `log_backtrace` too if you want the full backtrace of the exception.
We should definitely merge both functions.